scripts/format_notebooks.py

```diff
@@ -239,15 +258,59 @@ def preprocess_cell(self, cell, resources, cell_index):
             # Write notebook file
             if overwrite_notebook:
                 nb_file_loc = str(file_loc.with_suffix(""))
+            elif check_notebook:
+                _fd, nb_file_loc = tempfile.mkstemp()
             else:
                 nb_file_loc = str(file_loc.with_suffix(".mod"))
+
             print(f"Writing notebook to {nb_file_loc}.ipynb")
             writer.write(body, resources, nb_file_loc)
 
+            if check_notebook:
+                with open(file_loc) as f:
+                    original = f.read()
+
+                with open(nb_file_loc + ".ipynb") as f:
+                    updated = f.read()
+
+                if original != updated:
+                    check_failed.append(str(file_loc))
+
         if write_html:
             # Process the notebook to HTML
             (body, resources) = html_exporter.from_notebook_node(in_notebook)
 
             html_file_loc = str(file_loc.with_suffix(""))
             print(f"Writing HTML to {html_file_loc}.html")
             writer.write(body, resources, html_file_loc)
+
+    if check_failed:
+        assert check_notebook, "things failed check without check being enabled"
+
+        notebooks = "\n".join(f"- `{path}`" for path in check_failed)
+
+        command = "python ./scripts/format_notebooks.py --all --overwrite demos/"
+
+        message = f"""\
+Found notebook(s) with incorrect formatting:
+
+{notebooks}
+
+Fix by running:
+
+    {command}"""
+
+        print(f"\n\033[1;91;40mError:\033[0m {message}")
```

Comment:
May be a good idea defining these ANSI colours in a variable?

Reply:
Yeah, probably, can I do it as a separate PR, because the codes in the rest of the script should be factored out too?